### ISSUE TITLE
error with uturns parameter

### ISSUE THREAD

**Issue by vvk95** [Project-OSRM/osrm-backend#2047]
Hello! 

When parameter uturns=true, and there are no intersections, junctions and crossroads between two points, routes will be constructed as follows:
![uturn_error1](https://cloud.githubusercontent.com/assets/17332806/13487525/c016a65c-e12b-11e5-8c35-78d297ea3b72.jpg)

On another route the distance of detour has turned out more than 10 km.

**Comment by TheMarex** [comment]
Which version are you using? We fixed a problem like this recently on `develop`.

/cc @MoKob

**Comment by vvk95** [comment]
I used for test cases `http://router.project-osrm.org/`

For example:
1) `uturns=false`

Request:
`http://router.project-osrm.org/viaroute?instructions=true&loc=40.771751,-73.979959&loc=40.771011,-73.977202&loc=40.769768,-73.975260&loc=40.767347,-73.969510&output=json&uturns=false`

Response:
```
route_summary: {
total_distance: 1034,
total_time: 72,
end_point: "East 65th Street",
start_point: "West 65th Street"
},
```
![uturn_ny_false](https://cloud.githubusercontent.com/assets/17332806/13662920/13e7ed94-e6af-11e5-9bff-7f31b749981a.jpg)
Everything is good.

2) `uturns=true`

Request:
`http://router.project-osrm.org/viaroute?instructions=true&loc=40.771751,-73.979959&loc=40.771011,-73.977202&loc=40.769768,-73.975260&loc=40.767347,-73.969510&output=json&uturns=true`

Response:
```
route_summary: {
total_distance: 1527,
total_time: 111,
end_point: "East 65th Street",
start_point: "West 65th Street"
},
```
![uturn_ny_true](https://cloud.githubusercontent.com/assets/17332806/13663179/ce0ede0c-e6b0-11e5-9899-401f1ad85b9b.jpg)
A detour of 500 m.

@TheMarex  @MoKob

**Comment by TheMarex** [comment]
Confirmed that this still happens in the `rewrite/new-api` branch.

**Comment by TheMarex** [comment]
Should be fixed by e70bdc37016d410b3b2639f748534a309076692a
